adapters/epom/epom.go

```diff
@@ -0,0 +1,159 @@
+package epom
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the Epom adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) (requests []*adapters.RequestData, errors []error) {
+	rq, errs := a.makeRequest(request)
+
+	if len(errs) > 0 {
+		return nil, errs
+	}
+
+	if rq != nil {
+		requests = append(requests, rq)
+	}
+
+	return requests, nil
+}
+
+func (a *adapter) makeRequest(request *openrtb.BidRequest) (*adapters.RequestData, []error) {
+	var errs []error
+	var validImps []openrtb.Imp
+
+	if request.Device == nil || request.Device.IP == "" {
+		return nil, []error{&errortypes.BadInput{
+			Message: "No ip address in request",
+		}}
+	}
+
+	if request.Site == nil && request.App == nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: "Site or app field shouldn't be empty",
+		}}
+	}
+
+	for _, imp := range request.Imp {
+		if err := preprocess(&imp); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		validImps = append(validImps, imp)
+	}
+
+	if len(validImps) == 0 {
+		return nil, errs
+	}
+
+	request.Imp = validImps
+
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     a.endpoint,
+		Body:    reqJSON,
+		Headers: headers,
+	}, errs
+}
+
+func preprocess(imp *openrtb.Imp) error {
```

Comment:
I actually don't think you need this check. PBS-Core has you covered here too. See: https://github.com/prebid/prebid-server/blob/ef06fac6494f7e91f9d1d90ab07f02f8499b5efc/endpoints/openrtb2/auction.go#L472

Reply:
Ok. Here I removed id check and added check if impression media type is not audio (we don't support audio)